Makefile

```diff
@@ -67,6 +67,20 @@ vet: ## Vet the code and dependencies
 			exit 1;\
 		fi;}
 
+.PHONY: golangci-lint
+golangci-lint:
+# Run golangci-lint only from local machine
+	if [ "${CI}" != "true" ]; then \
+		if command -v golangci-lint &> /dev/null; then \
+			echo "Running golangci-lint"; \
+			golangci-lint run; \
+		else \
+			echo "Error: golangci-lint is not installed. Please run the 'make toolchain'"; \
+			exit 1; \
+		fi \
+	fi
```

Comment:
why we need CI conditional?

Reply:
In the new [presubmit.yaml](https://github.com/aws/aws-application-networking-k8s/blob/6fcdde8e913f85e92dcc40b4005e3c2f29a377cf/.github/workflows/presubmit.yaml), it has two jobs:
1. golangci-lint
2. presubmit
within the `presubmit` job , it does:
```
    - run: make toolchain
    - run: make presubmit (need to add a "${CI}" != "true" condition to prevent running golangci-lint)
    - run: make docs
    - name: Send coverage
```

Don't hope both the  `golangci-lint` github action job and `presubmit` github action job run the `golangci-lint run` two time, so we need a CI conditional

Reply:
Does not make sense to me, if it's in presubmit then it should not be in separate action, or remove it from presubmit.

Reply:
We need to use the `uses: golangci/golangci-lint-action@v3` github plugin  if its running as github action, this plugin has better integration will github, it could directly show report inline in the PR website UI: 
![image](https://github.com/aws/aws-application-networking-k8s/assets/32318664/cb62847f-814c-4148-920e-49af6320bf9b)

When people do local development, usually one step before people sending pull request is running `make presubmit`, people could see the   `golangci-lint` error message before send the PR

So, we need to config 2 way to run the golangci-lint.


Or, we could change people's habit to run bother `make lint` and  `make presubmit` before sending a PR?

Reply:
I'm fine with that, you add new make step that will include lint and presubmit only for local dev if you want to save extra command to run